csrc/runtime/executor_kernel_arg.cpp

```diff
@@ -46,21 +34,64 @@ PrimDataType getSmallestIndexType(const at::Tensor& tensor) {
 
 } // namespace
 
+void KernelArgumentHolder::push(const std::vector<at::Tensor>& tensors) {
+  for (const auto& tensor : tensors) {
+    arguments_.emplace_back(PolymorphicValue(tensor));
+  }
+}
+
 void KernelArgumentHolder::push(const c10::ArrayRef<c10::IValue>& args) {
-  // Naive I/O setup, I'm ignoring all the potential transformation (i.e. I/O
-  // allocated here from the subgraph could be, and very likely are, different
-  // from I/O expected by the generated CUDA kernel.
   for (const auto& arg : args) {
-    push(PolymorphicValue_functions::IValueToPolymorphicValue(arg));
+    arguments_.emplace_back(
+        PolymorphicValue_functions::IValueToPolymorphicValue(arg));
   }
 }
 
-void KernelArgumentHolder::push(const std::vector<at::Tensor>& tensors) {
-  for (const auto& tensor : tensors) {
-    push(tensor);
+void KernelArgumentHolder::push(const std::vector<c10::IValue>& args) {
+  for (const auto& arg : args) {
+    arguments_.emplace_back(
+        PolymorphicValue_functions::IValueToPolymorphicValue(arg));
   }
 }
 
+void KernelArgumentHolder::push(const at::Tensor& tensor) {
+  arguments_.emplace_back(PolymorphicValue(tensor));
+}
```

Comment:
Would it be better to do the following?
```suggestion
void KernelArgumentHolder::push(at::Tensor tensor) {
  arguments_.emplace_back(PolymorphicValue(std::move(tensor)));
}
```

This way, users can choose between the following two:

```C++
args.push(tensor); // do one copy in total
args.push(std::move(tensor)); // zero copy
```

where using const reference as parameter, there is no way to do zero-copying.

See also:
https://clang.llvm.org/extra/clang-tidy/checks/modernize/pass-by-value.html

Reply:
Yes. Updated.